### PR TITLE
Port to jbuilder and drop camlp4 library

### PR DESCRIPTION
**Pull request by mseri**
The latter is now in https://github.com/xapi-project/ocaml-rpc-legacy

### CODE REVIEW

**Comment by mseri** [comment]
I have just realised that I deleted the jbuilder port of tests and example. I'll get that back

File: CHANGES [hunk 1]
```diff
@@ -1,39 +1,3 @@
-3.2.0 (Mar 2018)
```
Comment (mseri) [inline CHANGES]: I need to checkout this from master :P

File: README.md [hunk 2]
```diff
@@ -5,8 +5,6 @@ using XML or JSON as transport encodings. The transport mechanism itself
 is outside the scope of this library as all conversions are from and to
 strings.
 
-[![Build Status](https://travis-ci.org/mirage/ocaml-rpc.svg?branch=master)](https://travis-ci.org/mirage/ocaml-rpc)
-
```
Comment (mseri) [inline README.md]: And this

File: rpc.opam [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+dev-repo: "git://github.com/mirage/ocaml-rpc"
+tags: [
+    "org:mirage"
+    "org:xapi-project"
+]
+build-test: [
+    ["jbuilder" "runtest" "-p" name]
+]
+depends: [
+    "jbuilder" {build}
```
Comment (mseri) [inline rpc.opam]: I can remove core and lwt from here now that rpclib-lwt and rpclib-async are in separate packages

File: rpclib.opam [hunk 4]
```diff
@@ -8,34 +8,25 @@ tags: [
...
-  "ocamlfind"
-  "ppx_deriving"
   "rresult"
-  "type_conv" {>= "108.07.01"}
   "xmlm"
   "yojson"
-  "camlp4"
 ]
-depopts: [ "js_of_ocaml" ]
+depopts: [
```
Comment (mseri) [inline rpclib.opam]: The rest here all goes now

**Comment by mseri** [comment]
Ping @gaborigloi, I have rebased it and solved the merge conflicts

**Comment by gaborigloi** [comment]
This needs to be rebased again :/

**Comment by mseri** [comment]
Rebased

File: rpc.opam (hunk 3 above)
Comment (gaborigloi) [reply rpc.opam]: I think it's a good idea to remove them.

**Comment by mseri** [comment]
Rebased, and integrated the new tests!

**Comment by mseri** [comment]
Discussed with @jonludlam, this will make it also easier to add further fixes and then backport them. Meging

**Pull request merged**